From: Vivien Nicolas <vnicolas@mozilla.com>

diff --git a/browser/app/profile/firefox.js b/browser/app/profile/firefox.js
--- a/browser/app/profile/firefox.js
+++ b/browser/app/profile/firefox.js
@@ -1409,8 +1409,12 @@ pref("browser.esedbreader.loglevel", "Er
 pref("browser.laterrun.enabled", false);
 
 // Enable browser frames for use on desktop.  Only exposed to chrome callers.
 pref("dom.mozBrowserFramesEnabled", true);
 
 pref("extensions.pocket.enabled", true);
 
 pref("extensions.webextensions.addon_implementation", true);
+pref("browser.cache.disk.enable", false);
+pref("browser.dom.window.dump.enabled", true);
+pref("toolkit.startup.max_resumed_crashes", -1);
+
