From: Vivien Nicolas <vnicolas@mozilla.com>

diff --git a/browser/components/BrowserUIProtocolHandler.js b/browser/components/BrowserUIProtocolHandler.js
--- a/browser/components/BrowserUIProtocolHandler.js
+++ b/browser/components/BrowserUIProtocolHandler.js
@@ -44,18 +44,17 @@ const Permissions = (function() {
     "browser:universal-xss"
   ];
 
   function add(uri, name) {
     Services.perms.add(uri, name, Ci.nsIPermissionManager.ALLOW_ACTION);
   }
 
   function remove(uri, name) {
-    let url = Services.io.newURI(uri.scheme + '://' + uri.host, null, null);
-    Services.perms.remove(url, name);
+    Services.perms.remove(uri, name);
   }
 
   return {
     set: function(uri) {
       kPermissions.forEach(function(name) { add(uri, name); });
     },
 
     unset: function(uri) {
